src/test/java/io/gingersnapproject/metrics/MetricsResourceTest.java

```diff
@@ -18,9 +20,9 @@
 import static org.hamcrest.CoreMatchers.containsString;
 
 @QuarkusTest
-@QuarkusTestResource(MySQLResources.class)
+@WithDatabase(value = MySQL.class, rules = MetricsResourceTest.RULE)
```

Comment:
So with this approach is is possible for us to execute this test against multiple DBs by providing `n` `@WithDatabase` annotations? e.g:

```
@QuarkusTest
@WithDatabase(value = MySQL.class, rules = MetricsResourceTest.RULE)
@WithDatabase(value = Postgres.class, rules = MetricsResourceTest.RULE)
```

Reply:
Looking at the annotation it is not possible currently. I tried to figure out how to do this with being able to reuse a single test class, but couldn't figure it out myself. I had to supply multiple test classes, one for each DB. I was thinking instead it might be better to use a system property to determine which DB it runs instead and then run maven N times for each DB.

But if anyone can figure out how to easily run with all the various DBs in a single run, I am all ears.

Reply:
Not right now, I think if we make the annotation repeatable, that should work. The problem is that the cache-manager container uses the same DB, so we need to either change the cache-manager to run with the different DBs or pin cache-manager with MySQL and start another container for the db-syncer if necessary.

Reply:
> The problem is that the cache-manager container uses the same DB, so we need to either change the cache-manager to run with the different DBs or pin cache-manager with MySQL and start another container for the db-syncer if necessary.

It seems like we need to utilise the per DB vendor cache-manager image(s) that will be provisioned by the Operator.

Reply:
@ryanemerson https://github.com/gingersnap-project/cache-manager/pull/37 adds support to have all DBs work with the same image. Let me know what you think of the approach. Or do you think we should go the single image per DB approach?

Reply:
> Or do you think we should go the single image per DB approach?

I think we definitely need this for production. It wasn't a priority before, as the focus was on functionality, but if it's going to simplify integration testing then maybe we should work on this sooner rather than later.

Reply:
So we have the repeatable annotation approach in commit https://github.com/gingersnap-project/db-syncer/pull/56/commits/68a6fc4f6e6948bc26c13c842d473d3d66e29f3d and the latest we have another using profiles.

By default, and to ease when running in the IDE, we default to MySQL. Then we run the tests again with the `postgres` profile.

If we want to run the test for a specific DB provider, we can then make it explicit in the `WithDatabase` annotation, and the test will run only for that provider once. If not specified, the test will run for all providers.